### PR TITLE
GitHub app auth

### PR DESCRIPTION
**Pull request by timja**
- Bump node version
- GitHub app auth
- Use GitHub checks instead of commit status

(Need to update infra first before merging to have the secrets etc)

Tested in:

- single module: https://github.com/jenkinsci/jenkins-infra-test-plugin/runs/4787014745
- multi module: https://github.com/jenkinsci/plugin-installation-manager-tool/runs/4795504430

Fixes https://github.com/jenkins-infra/incrementals-publisher/issues/3
Fixes https://github.com/jenkins-infra/incrementals-publisher/issues/4
Fixes https://github.com/jenkins-infra/helpdesk/issues/2426

### CODE REVIEW

File: lib/github.js [hunk 1]
```diff
@@ -22,18 +36,20 @@ module.exports = {
...
+    const github = await getRestClient();
+    return github.rest.checks.create({
       owner,
       repo,
-      sha,
-      state: 'success',
-      target_url,
-      description: 'Deployed to Incrementals.',
-      context: 'continuous-integration/jenkins/incrementals'
+      name: 'Incrementals',
```
Comment (halkeye) [inline lib/github.js:44]: this also looks like a change. Your changing from a commit status to a check?

**Review by halkeye — APPROVED** [review]
lgtm

File: lib/github.js (hunk 1 above)
Comment (timja) [reply lib/github.js:44]: Yup, so I can get multi modules giving all links, was horrid UX for jenkins core before and others you had to replace the first modules artifact id with the one you wanted in the URL, it never got the hpi / war

Comment (jglick) [reply lib/github.js:44]: I had filed an issue to this effect somewhere. Maybe only https://github.com/jenkins-infra/helpdesk/issues/2426.

File: lib/github.js [hunk 2]
```diff
@@ -22,18 +36,20 @@ module.exports = {
...
+      head_sha,
+      status: 'completed',
+      conclusion: 'success',
+      details_url: entries[0].url,
+      output: {
+        title: 'Deployed to Incrementals',
```
Comment (jglick) [inline lib/github.js:50]: I wonder if we should pick text that is more self-descriptive for people who do not already know what this means? Somehow indicating that this snapshot-ish build is available for download or consumption from Maven in draft PRs, linking to jenkins.io docs, etc.

File: lib/github.js [hunk 3]
```diff
@@ -22,18 +36,20 @@ module.exports = {
...
+        summary: entries.map(entry => `- [${entry.artifactId}](${entry.url})`).join('\n')
```
Comment (jglick) [inline lib/github.js:51]: And a good follow-up would be to show

```xml
<dependency>
    <groupId>org.jenkins-ci.plugins</groupId>
    <artifactId>stuff</artifactId>
    <version>1.2-rcdead1234beef</version>
</dependency>
```
```

**Review by jglick — COMMENTED** [review]
Nice!

File: lib/github.js (hunk 1 above)
Comment (jglick) [reply lib/github.js:44]: There we go, https://issues.jenkins.io/browse/JENKINS-52116 (CC @oleg-nenashev).

**Comment by timja** [comment]
FTR this is deployed but since the security release yesterday I haven't had a single pipeline build work, https://github.com/jenkins-infra/helpdesk/issues/2733

File: lib/github.js (hunk 3 above)
Comment (lemeurherve) [reply lib/github.js:51]: Addressed in #83 & #84

**Pull request merged**